Sources/AsyncHTTPClient/Utils.swift

```diff
@@ -32,3 +36,36 @@ public final class HTTPClientCopyingDelegate: HTTPClientResponseDelegate {
         return ()
     }
 }
+
+#if os(Linux)
+internal extension String {
+    var isIPAddress: Bool {
+      var ipv4Addr = in_addr()
+      var ipv6Addr = in6_addr()
+
+      return self.withCString { ptr in
+          return inet_pton(AF_INET, ptr, &ipv4Addr) == 1 ||
+                 inet_pton(AF_INET6, ptr, &ipv6Addr) == 1
+      }
+    }
+}
+#else
+internal extension String {
+    var isIPAddress: Bool {
+        if #available(OSX 10.14, *) {
+            if IPv4Address(self) != nil || IPv6Address(self) != nil {
+                return true
+            }
+        } else {
+            var ipv4Addr = in_addr()
+            var ipv6Addr = in6_addr()
+
+            return self.withCString { ptr in
+                return inet_pton(AF_INET, ptr, &ipv4Addr) == 1 ||
+                       inet_pton(AF_INET6, ptr, &ipv6Addr) == 1
+            }
+        }
```

Comment:
The fallback block is common across both operating systems, so it may be better to factor that out into a helper function that we can call directly to save repetition.